ssl/ssl_sess.c

```diff
@@ -575,7 +578,7 @@ int ssl_get_prev_session(SSL *s, CLIENTHELLO_MSG *hello)
         goto err;
     }
 
-    if (ret->timeout < (long)(time(NULL) - ret->time)) { /* timeout */
+    if (ret->timeout < (time(NULL) - ret->time)) { /* timeout */
```

Comment:
Aren't subtractions of two `time_t` values meant to be done using `difftime(3)` to avoid overflow issues?

Reply:
Yes, but man pages indicate `difftime(3)` could be defined as `(double)(t1 - t0)`, which doesn't offer protection.

`ret->time` is always less than or equal to `time(NULL)`, since `ret->time` is assigned a value of `time(NULL)`. 

The edge case might be when time of a machine is changed suddenly backwards a significant (more than a couple seconds). NTP adjustments wouldn't cause that probem.

Reply:
For that case, it be better to compare `ret->time` to "now", and if it is "in the future" assume there hasn't been a time out.

Reply:
I'm reluctant to use `difftime(3)` because of the floating point issues. And the *might* is very weak. But I think there may be a way some of these compares can be rewritten to always make it safe.